pagopa/io-app
# feat(Bonus Pagamenti Digitali): [#175973060] Change networking logic for bpd periods & amounts retrieval

## Short description
This pr changes the networking logic to retrieve bpd periods and amounts, in order to avoid to request amounts for inactive periods.

## List of changes proposed in this pull request
- Removed actions `bpdAmountLoad` and `bpdPeriodsLoad`, replaced with `bpdPeriodsAmountLoad`
- Renamed `BpdPeriodAmount` in `BpdPeriodWithAmount`
- Changed `bpdLoadAmountSaga`; now return `Either<BpdAmountError, BpdAmount>`
- Changed `bpdLoadPeriodsSaga`; now return `Either<Error, ReadonlyArray<BpdPeriod>>`
- Added new saga `loadPeriodsAmount` that uses `bpdLoadAmountSaga` and `bpdLoadPeriodsSaga` to compose the `ReadonlyArray<BpdPeriodWithAmount>`
- Removed `amount` from the store & reducer
- Now `periods` in the `bpd` store is a `pot.Pot<IndexedById<BpdPeriodWithAmount>, Error>`
- Now `selectedPeriod` in the `bpd` store is a `BpdPeriodWithAmount | null;`

## How to test
- Run `ts/features/bonus/bpd/saga/networking/__test__/loadPeriodsAmount.test.ts`
- Run the application with the mock data to see when is called the `bpdLoadAmountSaga`
